src/app/components/cards/TransferHistoryRow/index.jsx

```diff
@@ -0,0 +1,345 @@
+
+        /*  all transfers involve up to 2 accounts, context and 1 other. */
+        let message = '';
+
+        let description_start = '';
+        let other_account = null;
+        let description_end = '';
+
+        if (type === 'transfer_to_vesting') {
+            const amount = data.amount.split(' ')[0];
+
+            if (data.from === context) {
+                if (data.to === '') {
+                    message = tt(
+                        'transfer_history_row.transfer_to_vesting.from_self.no_to',
+                        { amount }
+                    );
+                    // tt('g.transfer') + amount + tt('g.to') + 'STEEM POWER';
+                } else {
+                    message = (
+                        <span>
+                            {tt(
+                                'transfer_history_row.transfer_to_vesting.from_self.to_someone',
+                                { amount }
+                            )}
+                            {otherAccountLink(data.to)}
+                        </span>
+                    );
+                    // tt('g.transfer') + amount + ' STEEM POWER' + tt('g.to');
+                }
+            } else if (data.to === context) {
+                message = (
+                    <span>
+                        {tt(
+                            'transfer_history_row.transfer_to_vesting.to_self',
+                            { amount }
+                        )}
+                        {otherAccountLink(data.from)}
+                    </span>
+                );
+                // tt('g.receive') + amount + ' STEEM POWER' + tt('g.from');
+            } else {
+                message = (
+                    <span>
+                        {tt(
+                            'transfer_history_row.transfer_to_vesting.from_user_to_user',
+                            {
+                                amount,
+                                from: data.from,
+                            }
+                        )}
+                        {otherAccountLink(data.to)}
+                    </span>
+                );
+                // tt('g.transfer') + amount + ' STEEM POWER' + tt('g.from') +data.from + tt('g.to');
+            }
+        } else if (
+            /^transfer$|^transfer_to_savings$|^transfer_from_savings$/.test(
+                type
+            )
+        ) {
+            // transfer_to_savings
+            const fromWhere =
+                type === 'transfer_to_savings'
+                    ? 'to_savings'
+                    : type === 'transfer_from_savings'
+                      ? 'from_savings'
+                      : 'not_savings';
+
+            if (data.from === context) {
+                // Semi-bad behavior - passing `type` to translation engine -- @todo better somehow?
+                // type can be to_savings, from_savings, or not_savings
+                // Also we can't pass React elements (link to other account) so its order is fixed :()
```

Comment:
Bummer. For future reference what library supports passing components?

Reply:
https://github.com/martinandert/react-interpolate-component along with https://github.com/martinandert/react-translate-component is one option that is sorta workable with our existing system

https://react.i18next.com is another one

my fave so far is https://github.com/lingui/js-lingui